src/test/shell/bazel/remote/remote_execution_test.sh

```diff
@@ -3771,4 +3776,108 @@ EOF
   expect_log "5 processes: 2 disk cache hit, 3 internal"
 }
 
+function test_unicode() {
+  # Bazel can handle Unicode paths if it has either a UTF-8 or ISO-8859-1
+  # locale available, but the in-tree remote execution worker has a hard
+  # requirement on UTF-8.
+  if ! "$is_windows"; then
```

Comment:
On Windows, locales aren't needed to decode file paths. The requirement of a UTF-8 locale is only applicable to UNIX platforms with arbitrary bytes in paths (Linux, BSD).

Reply:
I understand that Windows paths are Unicode, not arbitrary byte sequences. However, this test doesn't just deal in file *paths*, it also deals in file *contents* (namely, of .bzl files embedding non-ASCII characters).

Shouldn't we still test that Bazel will handle these files correctly *on Windows*? What happens if we remove `if ! $is_windows`? Is the Bash environment used by the integration tests complete enough to support `LC_ALL=en_US.UTF-8`?

Reply:
The Starlark spec requires UTF-8 encoding of `BUILD` and `.bzl` files:

https://github.com/bazelbuild/starlark/blob/master/spec.md#lexical-elements:
> A Starlark program consists of one or more modules. Each module is defined by a single UTF-8-encoded text file.

This has been a requirement since at least Bazel v0.17: https://docs.bazel.build/versions/0.17.1/skylark/spec.html#lexical-elements

I don't think Bazel enforces this, but from the discussion in https://github.com/bazelbuild/bazel/issues/4551 it sounds like a future version of Bazel might validate UTF-8 validity of inputs.

Reply:
I agree with everything you wrote, but it doesn't answer my question. I'm asking whether it's possible to run this test on Windows as well (where AIUI the "store UTF-8 as Latin1 and convert it back on the way out" hack still applies, even assuming BUILD/bzl files are UTF-8 encoded per the spec).

It's fine if the answer is "it's impossible / too hard to make this test run on Windows due to platform limitations". I won't insist on it then.

Reply:
I don't understand your question. This test is running on Windows.

Reply:
Nevermind, sorry. I misread the code.